### DISCUSSION TITLE
Call API after dbt model is finished

### DISCUSSION THREAD

**Discussion by kostek-pl** [dagster-io/dagster#18713] [Q&A] [answered]
Hello Everyone

I would like to trigger tableau workbook refreshment via api when dbt asset is finished(successfully).

What would be the best approach to do it ?

Make it as a downstream asset or sensor or maybe an op ?

Ideally would be to have one common stuff and make it dependent for selected models

**Comment by rexledesma** [comment, marked as answer]
If you know what models your tableau workbook is dependent on, then you can just create a downstream asset to your dbt models.

Something like:

```python
from dagster_dbt import get_asset_key_for_model
from dagster import asset

@asset(
    deps=[
        get_asset_key_for_model([my_dbt_assets], "my_dbt_model_1"),
        get_asset_key_for_model([my_dbt_assets], "my_dbt_model_2"),
        ...,
        get_asset_key_for_model([my_dbt_assets], "my_dbt_model_n"),
    ]
)
def my_downstream_tableau_workbook():
    ...
```

References:
- [Creating downstream dependencies to dbt models](https://docs.dagster.io/integrations/dbt/reference#downstream-dependencies)

**Reply by parkerdaminer** [reply]
Then you'd put the API call in the my_downstream_tableau_workbook()?

```python
def my_downstream_tableau_workbook():
    update_status = rest_api_call_to_tableau()
    return update_status
```

**Reply by rexledesma** [reply]
Correct.
